plugins/covid/models.py

```diff
@@ -42,3 +58,384 @@ class CovidPatient(models.Model):
+    active_malignancy_on_immunosuppression = models.NullBooleanField(
+        verbose_name='Active Malignancy On Immunosuppression (Includes radiotherapy)')
+    hiv                                    = models.NullBooleanField()
+    autoimmunne_with_immunosuppression     = models.NullBooleanField(
+        help_text='Autoimmune disease requiring current immunosuppression')
+    autoimmunne_without_immunosuppression  = models.NullBooleanField(
+        help_text='Autoimmune disease not requiring current immunosuppression')
+    gord                                   = models.NullBooleanField(verbose_name='GORD')
+    depression                             = models.NullBooleanField()
+    anxiety                                = models.NullBooleanField()
+    other_mental_health                    = models.NullBooleanField()
+    obesity                                = models.NullBooleanField()
+    dyslipidaemia                          = models.NullBooleanField()
+    anaemia                                = models.NullBooleanField()
+
+
+class CovidTrialEnrollment(EpisodeSubrecord):
+    """
+    Was this patient enrolled on a trial
+    """
+    _is_singleton = True
+    _icon         = 'fa fa-user-md'
+
+    enrolled = models.NullBooleanField()
+    details  = models.TextField(blank=True, null=True)
+
+
+class ITUAdmission(EpisodeSubrecord):
+    """
+    Details of ITU Admission
+    """
+    date_of_admission       = models.DateField(blank=True, null=True)
+    apache_score_on_arrival = models.CharField(blank=True, null=True, max_length=244, verbose_name='APACE score on ITU arrival')
+    intubated               = models.NullBooleanField()
+    date_of_intubation      = models.DateField(blank=True, null=True)
+
+
+class CovidCXR(EpisodeSubrecord):
+    """
+    Covid chest x ray
+    """
+    _icon = 'fa fa-crosshairs'
+
+    class Meta:
+        verbose_name = 'Covid CXR'
+
+    SEVERITY_CHOICES = enum(
+        'Mild',
+        'Moderate',
+        'Severe'
+    )
+
+    date = models.DateField(blank=True, null=True)
+    covid_code = models.CharField(blank=True, null=True, max_length=20, choices=COVID_CODE_CHOICES)
+
+
+class CovidCT(EpisodeSubrecord):
+    """
+    Covid CT Scans
+    """
+    _icon = 'fa fa-crosshairs'
+
+    class Meta:
+        verbose_name = 'Covid CT'
+
+    date   = models.DateField(blank=True, null=True)
+    pe     = models.NullBooleanField() # WHAT IS THIS? (Pulmonary_embolism)
+    report = models.TextField(blank=True, null=True)
+
+
+class CovidFollowUpCall(EpisodeSubrecord):
+    """
+    A phone call to a patient seeking follow up on their COVID-19 admission
+    """
+    _icon = 'fa fa-phone'
+
+    ONE_TO_TEN           = enum('1', '2', '3', '4', '5', '6', '7', '8', '9', '10')
+    ONE_TO_NINE          = enum('1', '2', '3', '4', '5', '6', '7', '8', '9')
+    ONE_TO_FIVE          = enum('1', '2', '3', '4', '5')
+    ZERO_TO_THREE        = (
```

Comment:
does enums with different fields definitely do what we think it does?

Reply:
? What do you mean ? What do you think it does?